### ISSUE TITLE
The `declaration-colon-newline-after` rule has no `never` option.

### ISSUE THREAD

**Issue by jtfairbank** [stylelint/stylelint#2060]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Feature Request: A new option to never have a newline after a declaration's colon.

> Which rule, if any, is this issue related to?

`declaration-colon-newline-after`

> What CSS is needed to reproduce this issue?

With the new option, the following are valid:

```css
.foo {
  border: 3px solid red;
}
```

```css
.foo {
  border: 3px
          solid
          red;
}
```

With the new option, the following are invalid:

```css
.foo {
  border:
    3px solid red;
}
```

```css
.foo {
  border:
    3px
    solid
    red;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "indentation": ["tab", {
      "except": ["value"]
    }]
  }
}
```

> Which version of stylelint are you using?

I'm using stylelint via the gulp-stylelint wrapper.  The versions of each are:

  * gulp-stylelint: `3.4.0`
  * stylelint: `7.4.2`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Using Gulp, via gulp-stylelint.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

I expected `declaration-colon-newline-after` to disallow newlines immediately after the colon, which could be combined with other value and declaration rules to enforce a particular declaration/value coding style.

> What actually happened (e.g. what warnings or errors you are getting)?

The option doesn't exist just yet. :)

**Comment by evilebottnawi** [comment]
sgtm for me

**Comment by davidtheclark** [comment]
@jtfairbank This is by design. http://stylelint.io/user-guide/about-rules/ in the "Rules work together" section tries to explain this. Looks to me like like what you want is `declaration-colon-space-after: "always"`, right? 

If you actually want to enforce *no* space after the colon, e.g. `color:pink`, you'd use `declaration-colon-space-after: "never"`.

**Comment by jtfairbank** [comment]
@davidtheclark You're absolutely right and I'm kinda dumb for not realizing that setting `declaration-colon-space-after: "always"` would enforce that.

On the downside, I just spend 2 hours figuring out the codebase and making test cases and a PR for this.  On the upside I can now contribute basic tests in the future.  Lucky you?

Just out of curiosity, how would I prevent something like this, where `declaration-colon-space-after` is met but there is a newline included after the space.  Basically, I want: `colon + 1 space + first value`.

```
.bad {
  color: /* space after is immediately followed by a newline */
    pink;
}

.worse {
  color:  pink; /* two spaces after... ew */
}

.good {
  color: pink; /* only 1 space between the colon and the first value */
}

.also-fine {
  border: 3px /* same, but it's ok if later values go on their own lines */
          solid
          pink;
}
```

**Comment by davidtheclark** [comment]
@jtfairbank If you can think of any way to clarify the documentation so other people don't suffer the same confusion, let us know!

> On the upside I can now contribute basic tests in the future.

That would be exciting :)

>  how would I prevent something like this, where declaration-colon-space-after is met but there is a newline included after the space.

The rule should already protect against this. It requires a *single* space after the colon.

**Comment by jtfairbank** [comment]
@davidtheclark

**Docs:**

1) It'd def be helpful to link to the [about rules page](http://stylelint.io/user-guide/about-rules/) from the top of the [rules page](http://stylelint.io/user-guide/rules/).  I did read the User Guide and the "rules work together" section, but I could see how a lot of newbies like me could miss it.  Or experienced users who just haven't setup rules in a while and are starting a new project.

2) It'd be a helpful reminder to link to the "Rules Work Together" section from various rule pages that you expect to be used in this way.  It's too easy to lookup at each rule individually (or be linked there from some code comment), and that means you aren't taking a holistic view of the rules and how to use them together.  Even better: link to specific examples in the "Rules Work Together" section for the rules in those examples and their cousins.

3) A list of links to related rules would be awesome, both in terms of providing context / a reminder to use them together, and just in terms of navigating around without having to go back and forth all the time.

**Tests:**

Any suggestions on where to start / which rules or options need the most help?

**Single Space Req:**

Ok, good to know.  FYI- that wasn't immediately obvious to me b/c it wasn't highlighted as a warning example on the [declaration-colon-space-after docs](http://stylelint.io/user-guide/rules/declaration-colon-space-after/).

How come you can't have an "extended examples" page for each rule which is automatically built from the test cases themselves?

**Overall:**

Really cool project.  Besides some thoughtful nit-picking in this bug report, I have nothing major to complain about.

And now I can force my css source styling will upon the code monkeys doing my evil bidding!  jk... they kept mashing on the keyboards until they invented Facebook so I haven't seen em since '06.  (oooh burn Zuck)

**Comment by davidtheclark** [comment]
@jtfairbank Thanks. I created #2066 and #2067 based on these suggestions.

Regarding tests: You can always feel free to pick up one of the open issues, create a feature or fix a bug and add tests for your change.
